presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -98,6 +108,12 @@
             .build(CacheLoader.from(key ->
                     internalCompileHashStrategy(key.getTypes(), key.getOutputChannels(), key.getJoinChannels(), key.getSortChannel())));
 
+    private ScalarFunctionImplementation getOperator(Type key)
```

Comment:
This method name is too vague.

Also, why don't we follow the pattern of `typeEqualsIgnoreNulls` and `typeEquals` to have `typeIsNotDistinctFrom` method, and returns the `BytecodeNode` ? Did I miss any discussion?